packages/wp-story-editor/src/components/corsCheck/corsCheck.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback, useState } from '@googleforcreators/react';
+import { useAPI } from '@googleforcreators/story-editor';
+import { useFeature } from 'flagged';
+
+/**
+ * Internal dependencies
+ */
+import CorsCheckFailed from './corsCheckFailed';
+
+function CorsCheck() {
+  const [showDialog, setShowDialog] = useState(false);
+  const [doCheck, setDoCheck] = useState(true);
```

Comment:
If you use an empty dependencies array for the `useEffect` below (and suppress the lint error for it), then this state is not needed, because the effect will only run once.